core/src/main/java/io/opencensus/tags/TagContext.java

```diff
@@ -47,30 +49,39 @@
   }
 
   /**
-   * Returns a builder based on this {@code TagContext}.
+   * Returns the current {@code TagContext}.
    *
-   * @return a builder based on this {@code TagContext}.
+   * @return the current {@code TagContext}.
    */
-  public Builder toBuilder() {
-    return new Builder(getTags());
+  public static TagContext getCurrentTags() {
```

Comment:
That's a great idea.  I didn't consider the benefit of avoiding building the map of tags when it's not needed.  Now I'm convinced that we should use an abstract factory for TagContexts, too.   I'll update this branch and make a new PR.